invenio_rdm_records/services/communities/service.py

```diff
@@ -6,24 +6,100 @@
+    ExceedMaxNumberOfCommunities,
+    RecordCommunityMissing,
+)
 
 
-class RecordCommunitiesService(RecordService):
+class RecordCommunitiesService(Service):
     """Record communities service.
 
     The communities service is in charge of managing communities of a given record.
     """
 
+    @property
+    def indexer(self):
+        """Factory for creating an indexer instance."""
+        return self.config.indexer_cls(
+            # the routing key is mandatory in the indexer constructor since
+            # it is afterwards passed explicitly to the created consumers
+            # and producers. this means that it is not strictly necessary to
+            # pass it to the queue constructor. however, it is passed for
+            # consistency (in case the queue is used by itself) and to help
+            # entity declaration on publish.
+            queue=LocalProxy(
+                lambda: Queue(
+                    self.config.indexer_queue_name,
+                    exchange=current_app.config["INDEXER_MQ_EXCHANGE"],
+                    routing_key=self.config.indexer_queue_name,
+                )
+            ),
+            routing_key=self.config.indexer_queue_name,
+            record_cls=self.config.record_cls,
+            record_to_index=self.record_to_index,
+            record_dumper=self.config.index_dumper,
+        )
+
+    def record_to_index(self, record):
+        """Function used to map a record to an index."""
+        return record.index._name
+
+    @property
+    def schema(self):
+        """Returns the data schema instance."""
+        return ServiceSchemaWrapper(self, schema=self.config.schema)
+
+    @property
+    def record_cls(self):
+        """Factory for creating a record class."""
+        return self.config.record_cls
+
     @unit_of_work()
     def create(self, identity, data, record, uow=None):
         """Include the record in the given communities."""
         pass
 
+    def _remove(self, community_id, record):
+        """Remove a community from the record."""
+        if community_id not in record.parent.communities.ids:
+            raise RecordCommunityMissing(record.id, community_id)
+
+        record.parent.communities.remove(community_id)
+
     @unit_of_work()
-    def delete(self, identity, id_, revision_id=None, uow=None):
+    def delete(self, identity, record_id, data, revision_id=None, uow=None):
         """Remove communities from the record."""
-        pass
+        record = self.record_cls.pid.resolve(record_id)
+        self.require_permission(identity, "remove_community", record=record)
+
+        valid_data, errors = self.schema.load(
+            data,
+            context={"identity": identity},
+            raise_errors=True,
+        )
+        communities = valid_data["communities"]
+        if self.config.max_number_of_removals < len(communities):
+            raise ExceedMaxNumberOfCommunities(self.config.max_number_of_removals)
```

Comment:
sorry for being late to the party, but just to understand why we introduce this contraint.  Why we cannot remove more than configured ?

Reply:
To avoid potential timeouts when removing communities. Every time we remove a community we have to commit the parent record and index the record, one by one. In this case - removing communities from a record - it's very unlikely that the limit will be reached, but when removing records that belong to a community, it will be more plausible to delete a big number of records  (a community can have thousands of records).

Reply:
didn't we say to use celery instead because of timeouts?

Reply:
IIRC we agreed to use celery only when deleting a community, here the task: https://github.com/inveniosoftware/invenio-communities/issues/891. 

In this case it's slightly different as it's the user who selects the communities that (s)he wants to remove. This way it's easier to report back the errors to the user and it's more user-friendly as well, as they can see the outcome of the action inmediately.